extensions/banner/deployment/src/main/java/io/quarkus/banner/deployment/BannerProcessor.java

```diff
@@ -0,0 +1,53 @@
+package io.quarkus.banner.deployment;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.quarkus.banner.runtime.BannerRecorder;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.HotDeploymentWatchedFileBuildItem;
+import io.quarkus.deployment.util.FileUtil;
+
+public class BannerProcessor {
+
+    private static final String DEFAULT_BANNER_PATH = "src/main/resources/banner.txt";
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public void recordBanner(BannerRecorder recorder, BannerConfig config) {
+        String content = readBannerFile(config.path);
+        recorder.print(content);
+    }
+
+    @BuildStep
+    List<HotDeploymentWatchedFileBuildItem> watchBannerChanges(BannerConfig config) {
```

Comment:
Since we are returning a list containing only one item the piece of code could be refactored to 

```
@BuildStep
HotDeploymentWatchedFileBuildItem watchBannerChanges(BannerConfig config) {
        return new HotDeploymentWatchedFileBuildItem(config.path);
    }
```